common/common.go

```diff
@@ -653,3 +654,29 @@ func GetTypeAssertError(required string, received interface{}, fieldDescription
 	}
 	return fmt.Errorf("%w from %T to %s%s", ErrTypeAssertFailure, received, required, description)
 }
+
+// Batch takes a slice type and converts it into a slice of slices
+func Batch[S ~[]E, E any](blobs S, batchSize int) []S {
+	if batchSize <= 0 || len(blobs) == 0 {
+		return []S{}
```

Comment:
> but you've put a batchSize == 0 check for Kucoin
I didn't understand this.

It feels to me like if someone writes Batch(b, 0) then they wouldn't be any more correct to expect a batchSize of blobs than an empty resultset.
I'm leaning heavily towards leaving this as-it-is because returning a slice with an empty slice in it feels even more wrong.

Reply:
```go
output := Batch(inputSlice, maxAmountCanBeZero)
```

```go
output := []type{inputSlice}
if  maxAmountCannotBeZero > 0 {
output = Batch(inputSlice, maxAmountCannotBeZero )
}
```

I see both scenarios being valid. Look at dem lines saved though 😆.

Reply:
> I'm leaning heavily towards leaving this as-it-is because returning a slice with an empty slice in it feels even more wrong

For a scenario where len(blob) > 0 and batch is a var that happens to be zero, one gets an empty batch, must check the batch, then use blob instead. That's the only scenario I didn't like. I'm fine with it staying

Reply:
> where you need to put checking code around the function call like you do with Kucoin
Can you point to that code?

Cos I don't think I have any that does that.
The only thing we have is the (documented) switch that will make it use `ticker:all` if you have more than 10 pairs enabled.

Reply:
TL;DR:
* Yes, `batchSize <= 0` should act like len(blobs) as suggested.
* *But* len(blobs) 0 should still produce an empty slice, not a slice with one empty slice in it

### Discussion
That really comes down to what you think a batch size of 0 should do.
A batch size of 1 means: Give me slices with 1 elements in each slice.
My first thought was: Therefore batch size of 0 should not swing completely the other direction and act as batchsize is len(blobs).
The logically correct behaviour feels like:
> You asked for batches with zero elements. Here's the empty slice you asked for.

But then I thought: Why would the batchSize be dynamic?
Most likely, because it's from config.
And if it's from config, what does an empty batch size mean?
( Or even a negative one, in muggle-minded people? )
It would mean "don't batch it", which supports the requested change.

Reply:
Fixed 85567a25

Added behavioural tests in case any of this regresses.
Clarified function comment, since it was not clear if batchSize is the size of the batches or the number of batches.